strider/fetcher.py

```diff
@@ -272,14 +359,30 @@ async def generate_from_kp(
     async def generate_from_result(
         self,
         qgraph,
+        key_fcn,
         get_results: Callable[[dict], Iterable[tuple[dict, dict]]],
+        result_map,
         call_stack: List,
+        sub_qid: str,
     ):
-        async for subkgraph, subresult, subauxgraph in self.lookup(
+        # self.logger.error(f"[{qid}] Result map: {result_map.keys()}")
+        async for subkgraph, subresult, subauxgraph, qid in self.lookup(
             qgraph,
             call_stack,
+            sub_qid,
         ):
+            self.logger.debug(
+                f"[{qid}] looking for key {key_fcn(subresult)}: {subresult.json()}"
+            )
+            # final hop yield should be empty result
+            if not key_fcn(subresult) in result_map:
+                self.logger.error(
+                    f"[{qid}] Couldn't find subresult in result map: {key_fcn(subresult)}"
+                )
+                self.logger.error(f"[{qid}] subresult from lookup: {subresult.json()}")
+                raise KeyError("Subresult not found in result map")
```

Comment:
Instead of erroring out, would it be possible to just log the error and continue, such that if something like this happens, we still get results from the other callstacks?

Reply:
I'm worried that if we don't do something drastic, we won't end up checking the logs to see this error. We never want this dropping of results to happen.